gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -591,11 +602,33 @@ private void startAtLatestOffset() {
     @Override
     public void run() {
       try (Timer.Context context = metricContext.timer(WORK_UNITS_FOR_TOPIC_TIMER).time()) {
+        // use shared client if configure, otherwise create a thread local one
+        if (KafkaSource.this.sharedKafkaConsumerClient != null) {
+          KafkaSource.this.kafkaConsumerClient.set(KafkaSource.this.sharedKafkaConsumerClient);
+        } else {
+          KafkaSource.this.kafkaConsumerClient.set(
```

Comment:
Isn't this essentially a local variable at this point? You're not reusing it within the thread, every `WorkUnitCreator` creates a new one.